### PR TITLE
fix(broker): pick latest snapshot on load when multiple snapshots exists

### LINKED ISSUE #5500: IllegalStateException: Expected to find event with the snapshot position ...
**Describe the bug**
Observed as error in production logs.

**Log/Stacktrace**
https://console.cloud.google.com/errors/CKnfqtj3lqTxaw?service=zeebe&time=P7D&project=camunda-cloud-240911

**Environment:**
Camunda Cloud / Prod / 0.24.3

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

- Pick latest snapshot on startup when multiple snapshot exists
- Delete snapshot directory if atomic move fails. If snapshot was partially moved, it would be considered a valid snapshot after a restart.

## Related issues

closes #5500 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStore.java [hunk 1]
```diff
@@ -71,7 +71,14 @@ private FileBasedSnapshot loadLatestSnapshot(final Path snapshotDirectory) {
     FileBasedSnapshot latestPersistedSnapshot = null;
     try (final var stream = Files.newDirectoryStream(snapshotDirectory)) {
       for (final var path : stream) {
-        latestPersistedSnapshot = collectSnapshot(path);
+        final var snapshot = collectSnapshot(path);
+        if (latestPersistedSnapshot == null) {
+          latestPersistedSnapshot = snapshot;
+        } else if (snapshot != null
```
Comment (MiguelPires) [inline snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStore.java:77]: Couldn't we delete the snapshot here if it's less recent than the `latestPersistedSnapshot`?

Comment (deepthidevaki) [reply snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStore.java]: Yes. I fixed it now. I now refactored it to make it explicit that we are deleting all older snapshot.

**Review by miguelpires — APPROVED** [review]
Looks good :+1:

**Pull request merged**